### DISCUSSION TITLE
Need help with listing enterprise members either through REST API or GraphQL

### DISCUSSION THREAD

**Discussion by RInkeshIdx9** [community/community#64830] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Hi,
I am working on integrating GitHub to SAAS Application where I need to grab all the members, owners and organizations of an enterprise. I did some research around REST API for the same but was not able to find one. Also, I am trying to get it through GraphQL but their is some error saying `Selection can't be made directly on unions.

Can somebody help me with either pointing to APIs that give this data or else a GraphQL query.

Any help would be appreciated.
Thanks in advance.

**Comment by arafajs** [comment]
You can use the following endpoints to get the information you need:

- `GET /orgs/:org/members` to get a list of all the members of an organization.
- `GET /orgs/:org/owners` to get a list of all the owners of an organization.
- `GET /orgs` to get a list of all the organizations that a user belongs to.

**Reply by RInkeshIdx9** [reply]
Thanks @arafajs but I am looking for enterprise's members and owners and not organization's

**Reply by arafajs** [reply]
Sorry. Here is the GraphQL query to get all the members and owners of an enterprise:

```
query {
  enterprise {
    members {
      id
      login
    }
    owners {
      id
      login
    }
  }
}
```
This query will first get the enterprise object. Then, it will get the members and owners fields of the enterprise object. These fields will return a list of all the members and owners of the enterprise, respectively.

**Reply by RInkeshIdx9** [reply]
This is giving me error
`{
  "errors": [
    {
      "path": [
        "query",
        "enterprise",
        "members",
        "id"
      ],
      "extensions": {
        "code": "undefinedField",
        "typeName": "EnterpriseMemberConnection",
        "fieldName": "id"
      },
      "locations": [
        {
          "line": 4,
          "column": 7
        }
      ],
      "message": "Field 'id' doesn't exist on type 'EnterpriseMemberConnection'"
    },
    {
      "path": [
        "query",
        "enterprise",
        "members",
        "login"
      ],
      "extensions": {
        "code": "undefinedField",
        "typeName": "EnterpriseMemberConnection",
        "fieldName": "login"
      },
      "locations": [
        {
          "line": 5,
          "column": 7
        }
      ],
      "message": "Field 'login' doesn't exist on type 'EnterpriseMemberConnection'"
    },
    {
      "path": [
        "query",
        "enterprise",
        "owners"
      ],
      "extensions": {
        "code": "undefinedField",
        "typeName": "Enterprise",
        "fieldName": "owners"
      },
      "locations": [
        {
          "line": 7,
          "column": 5
        }
      ],
      "message": "Field 'owners' doesn't exist on type 'Enterprise'"
    }
  ]
}`

**Reply by RInkeshIdx9** [reply]
Because members is of type EnterpriseMemberConnection which does not have id or login it only has edges or nodes

**Reply by arafajs** [reply]
```
query {
  enterprise {
    members {
      nodes {
        id
        login
      }
    }
    owners {
      nodes {
        id
        login
      }
    }
  }
}
```

**Reply by RInkeshIdx9** [reply]
Nope this also didn't work

**Reply by arafajs** [reply]
and this?
```
query {
  enterprise {
    organizations {
      nodes {
        id
        login
        members {
          nodes {
            id
            login
          }
        }
        owners {
          nodes {
            id
            login
          }
        }
      }
    }
  }
}
```

**Reply by RInkeshIdx9** [reply]
This worked

**Reply by RInkeshIdx9** [reply, marked as answer]
But with some tweaks
```
query {
  enterprise(slug: "slug") {
    organizations(first: 100) {
      nodes {
        id
        login
        membersWithRole(first: 100){
          nodes{
            login
          }
        }
        enterpriseOwners(first: 100){
          nodes{
            login
          }
        }
      }
    }
  }
}
```

**Reply by RInkeshIdx9** [reply]
Is their any way we could get billing managers
